### PR TITLE
Use parallel builds to speed up compilation

### LINKED ISSUE #44: Use parallel builds
`hammer.bat` supports parallel builds with the `-j` option. E.g. `hammer.bat -j32` for building with 32 cores. This significantly speeds up build times. The goal of this issue is to make `build_omaha.py` pass a suitable `-j` value to `hammer.bat` when building, and thus to likewise speed up Brave's build (`create_dist --build_omaha`) on Windows.

### PR DESCRIPTION
**Pull request by mherrmann**
Resolves #44.

### CODE REVIEW

File: build_omaha.py [hunk 1]
```diff
@@ -22,7 +23,8 @@ def build(omaha_dir, standalone_installers_dir, build_all):
   mode = 'opt-win'
   if build_all:
     mode = 'all'
-  command = ['hammer.bat', 'MODE=' + mode, '--all', '--standalone_installers_dir=' + standalone_installers_dir]
+  command = ['hammer.bat', 'MODE=' + mode, '--all', '--standalone_installers_dir=' + standalone_installers_dir,
+             '-j%d' % multiprocessing.cpu_count()]
```
Comment (mihaiplesa) [inline build_omaha.py:27]: Do we want this to be passed from outside instead?

Comment (mherrmann) [reply build_omaha.py:27]: I don't know for sure and would say the answer depends on whether this ability - to change the `-j` value from outside - will actually be used in practice. Or whether it's a general suggested pattern in the build to only ever take such values from outside. As it is, the advantage is that it's very simple.

**Review by mihaiplesa — APPROVED** [review]

**Review by atuchin-m — APPROVED** [review]

**Comment by atuchin-m** [comment]
Could we check this on a test build before merging?

**Comment by mherrmann** [comment]
@atuchin-m: I kicked one off with these changes in https://github.com/brave/brave-core/pull/12557

**Review by simonhong — APPROVED** [review]
When I porked omaha to brave (maybe 4y ago),
I didn't use parallel build because I saw some build failure on my local machine with `-j` option.
Happy to see parallel build for omaha! :)

**Comment by mherrmann** [comment]
@simonhong thank you for the review! My suspicion is that the build failures were/are due to missing Python dependency `pywin32`. See https://github.com/brave/devops/issues/7101.

**Pull request merged**